skaha/src/main/java/org/opencadc/skaha/session/PostAction.java

```diff
@@ -244,28 +261,127 @@ public void doAction() throws Exception {
-                
+
                 // gather job parameters
                 String image = syncInput.getParameter("image");
-                List<String> params = syncInput.getParameters("param");
-                
+
                 if (image == null) {
                     throw new IllegalArgumentException("Missing parameter 'image'");
                 }
-                
-                attachDesktopApp(image, params);
-                
+
+                attachDesktopApp(image);
+
             } else {
                 throw new UnsupportedOperationException("Cannot modify an existing app.");
             }
         }
     }
-    
+
+    void ensureUserBase() throws IOException, InterruptedException {
+        final Path homeDir = Paths.get(String.format("%s/%s", this.homedir, this.userID));
+
+        if (Files.notExists(homeDir)) {
+            allocateUser();
+        }
+    }
+
+    void allocateUser() throws IOException, InterruptedException {
+        log.debug("PostAction.makeUserBase()");
+        final ProcessBuilder processBuilder = new ProcessBuilder().command(PostAction.MAKE_USER_BASE_COMMAND);
+        final Process p = processBuilder.start();
+        final String jobName = K8SUtil.getJobName(this.sessionID, PostAction.SKAHA_ADD_USER_TYPE, this.userID);
+        try (final BufferedOutputStream commandInput = new BufferedOutputStream(p.getOutputStream())) {
+            processCommandInput(commandInput, jobName);
+            commandInput.flush();
+        }
+
+        log.debug("Executing " + Arrays.toString(PostAction.MAKE_USER_BASE_COMMAND));
+        final int code = p.waitFor();
+        try (final InputStream stdOut = new BufferedInputStream(p.getInputStream());
+             final InputStream stdErr = new BufferedInputStream(p.getErrorStream())) {
+            if (code != 0) {
+                final String errorOutput = readStream(stdErr);
+                final String commandOutput = readStream(stdOut);
+                throw new IOException("Unable to create user home (code: " + code + ")."
+                                      + "\nError message from server: " + errorOutput
+                                      + "\nOutput from command: " + commandOutput);
+            } else {
+                log.debug(readStream(stdOut));
+                log.debug("Executing " + Arrays.toString(PostAction.MAKE_USER_BASE_COMMAND) + ": OK");
+            }
+        }
+
+        // Needs to match what is in the YAML job file.
+        waitForUserAllocation(jobName);
+
+        log.debug("PostAction.makeUserBase(): OK");
+    }
+
+    /**
+     * Ensure user allocation is complete.  This is essential before progressing to Session Creation.
+     * @throws IOException              If the process cannot execute.
+     * @throws InterruptedException     If waiting for the process cannot happen or is interrupted.
+     */
+    void waitForUserAllocation(final String jobName) throws IOException, InterruptedException {
```

Comment:
I think we need a max wait time on this.  2 minutes?  If reached an IllegalStateException should be thrown.